// @aws-sdk/client-elastic-load-balancing-v2/dist-es/commands/AddListenerCertificatesCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { AddListenerCertificatesInput, AddListenerCertificatesOutput } from "../models/models_0";
import { deserializeAws_queryAddListenerCertificatesCommand, serializeAws_queryAddListenerCertificatesCommand, } from "../protocols/Aws_query";
var AddListenerCertificatesCommand = (function (_super) {
    __extends(AddListenerCertificatesCommand, _super);
    function AddListenerCertificatesCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    AddListenerCertificatesCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ElasticLoadBalancingV2Client";
        var commandName = "AddListenerCertificatesCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: AddListenerCertificatesInput.filterSensitiveLog,
            outputFilterSensitiveLog: AddListenerCertificatesOutput.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    AddListenerCertificatesCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryAddListenerCertificatesCommand(input, context);
    };
    AddListenerCertificatesCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryAddListenerCertificatesCommand(output, context);
    };
    return AddListenerCertificatesCommand;
}($Command));
export { AddListenerCertificatesCommand };
