// @aws-sdk/client-elastic-load-balancing-v2/dist-es/commands/CreateTargetGroupCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { CreateTargetGroupInput, CreateTargetGroupOutput } from "../models/models_0";
import { deserializeAws_queryCreateTargetGroupCommand, serializeAws_queryCreateTargetGroupCommand, } from "../protocols/Aws_query";
var CreateTargetGroupCommand = (function (_super) {
    __extends(CreateTargetGroupCommand, _super);
    function CreateTargetGroupCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    CreateTargetGroupCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ElasticLoadBalancingV2Client";
        var commandName = "CreateTargetGroupCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: CreateTargetGroupInput.filterSensitiveLog,
            outputFilterSensitiveLog: CreateTargetGroupOutput.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    CreateTargetGroupCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryCreateTargetGroupCommand(input, context);
    };
    CreateTargetGroupCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryCreateTargetGroupCommand(output, context);
    };
    return CreateTargetGroupCommand;
}($Command));
export { CreateTargetGroupCommand };
