// @aws-sdk/client-elastic-load-balancing-v2/dist-es/commands/CreateLoadBalancerCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { CreateLoadBalancerInput, CreateLoadBalancerOutput } from "../models/models_0";
import { deserializeAws_queryCreateLoadBalancerCommand, serializeAws_queryCreateLoadBalancerCommand, } from "../protocols/Aws_query";
var CreateLoadBalancerCommand = (function (_super) {
    __extends(CreateLoadBalancerCommand, _super);
    function CreateLoadBalancerCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    CreateLoadBalancerCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ElasticLoadBalancingV2Client";
        var commandName = "CreateLoadBalancerCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: CreateLoadBalancerInput.filterSensitiveLog,
            outputFilterSensitiveLog: CreateLoadBalancerOutput.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    CreateLoadBalancerCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryCreateLoadBalancerCommand(input, context);
    };
    CreateLoadBalancerCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryCreateLoadBalancerCommand(output, context);
    };
    return CreateLoadBalancerCommand;
}($Command));
export { CreateLoadBalancerCommand };
