// @aws-sdk/client-elastic-load-balancing-v2/dist-es/commands/DeleteTargetGroupCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DeleteTargetGroupInput, DeleteTargetGroupOutput } from "../models/models_0";
import { deserializeAws_queryDeleteTargetGroupCommand, serializeAws_queryDeleteTargetGroupCommand, } from "../protocols/Aws_query";
var DeleteTargetGroupCommand = (function (_super) {
    __extends(DeleteTargetGroupCommand, _super);
    function DeleteTargetGroupCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DeleteTargetGroupCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ElasticLoadBalancingV2Client";
        var commandName = "DeleteTargetGroupCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DeleteTargetGroupInput.filterSensitiveLog,
            outputFilterSensitiveLog: DeleteTargetGroupOutput.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DeleteTargetGroupCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryDeleteTargetGroupCommand(input, context);
    };
    DeleteTargetGroupCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryDeleteTargetGroupCommand(output, context);
    };
    return DeleteTargetGroupCommand;
}($Command));
export { DeleteTargetGroupCommand };
