// @aws-sdk/client-elastic-load-balancing-v2/dist-es/commands/DeleteListenerCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DeleteListenerInput, DeleteListenerOutput } from "../models/models_0";
import { deserializeAws_queryDeleteListenerCommand, serializeAws_queryDeleteListenerCommand, } from "../protocols/Aws_query";
var DeleteListenerCommand = (function (_super) {
    __extends(DeleteListenerCommand, _super);
    function DeleteListenerCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DeleteListenerCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ElasticLoadBalancingV2Client";
        var commandName = "DeleteListenerCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DeleteListenerInput.filterSensitiveLog,
            outputFilterSensitiveLog: DeleteListenerOutput.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DeleteListenerCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryDeleteListenerCommand(input, context);
    };
    DeleteListenerCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryDeleteListenerCommand(output, context);
    };
    return DeleteListenerCommand;
}($Command));
export { DeleteListenerCommand };
