// @aws-sdk/client-elastic-load-balancing-v2/dist-es/commands/DescribeSSLPoliciesCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DescribeSSLPoliciesInput, DescribeSSLPoliciesOutput } from "../models/models_0";
import { deserializeAws_queryDescribeSSLPoliciesCommand, serializeAws_queryDescribeSSLPoliciesCommand, } from "../protocols/Aws_query";
var DescribeSSLPoliciesCommand = (function (_super) {
    __extends(DescribeSSLPoliciesCommand, _super);
    function DescribeSSLPoliciesCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DescribeSSLPoliciesCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ElasticLoadBalancingV2Client";
        var commandName = "DescribeSSLPoliciesCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DescribeSSLPoliciesInput.filterSensitiveLog,
            outputFilterSensitiveLog: DescribeSSLPoliciesOutput.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DescribeSSLPoliciesCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryDescribeSSLPoliciesCommand(input, context);
    };
    DescribeSSLPoliciesCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryDescribeSSLPoliciesCommand(output, context);
    };
    return DescribeSSLPoliciesCommand;
}($Command));
export { DescribeSSLPoliciesCommand };
