// @aws-sdk/client-elastic-load-balancing-v2/dist-es/commands/DescribeListenerCertificatesCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DescribeListenerCertificatesInput, DescribeListenerCertificatesOutput } from "../models/models_0";
import { deserializeAws_queryDescribeListenerCertificatesCommand, serializeAws_queryDescribeListenerCertificatesCommand, } from "../protocols/Aws_query";
var DescribeListenerCertificatesCommand = (function (_super) {
    __extends(DescribeListenerCertificatesCommand, _super);
    function DescribeListenerCertificatesCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DescribeListenerCertificatesCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ElasticLoadBalancingV2Client";
        var commandName = "DescribeListenerCertificatesCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DescribeListenerCertificatesInput.filterSensitiveLog,
            outputFilterSensitiveLog: DescribeListenerCertificatesOutput.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DescribeListenerCertificatesCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryDescribeListenerCertificatesCommand(input, context);
    };
    DescribeListenerCertificatesCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryDescribeListenerCertificatesCommand(output, context);
    };
    return DescribeListenerCertificatesCommand;
}($Command));
export { DescribeListenerCertificatesCommand };
