// @aws-sdk/client-elastic-load-balancing-v2/dist-es/commands/DescribeListenersCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DescribeListenersInput, DescribeListenersOutput } from "../models/models_0";
import { deserializeAws_queryDescribeListenersCommand, serializeAws_queryDescribeListenersCommand, } from "../protocols/Aws_query";
var DescribeListenersCommand = (function (_super) {
    __extends(DescribeListenersCommand, _super);
    function DescribeListenersCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DescribeListenersCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ElasticLoadBalancingV2Client";
        var commandName = "DescribeListenersCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DescribeListenersInput.filterSensitiveLog,
            outputFilterSensitiveLog: DescribeListenersOutput.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DescribeListenersCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryDescribeListenersCommand(input, context);
    };
    DescribeListenersCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryDescribeListenersCommand(output, context);
    };
    return DescribeListenersCommand;
}($Command));
export { DescribeListenersCommand };
