// @aws-sdk/client-elastic-load-balancing-v2/dist-es/commands/ModifyTargetGroupCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ModifyTargetGroupInput, ModifyTargetGroupOutput } from "../models/models_0";
import { deserializeAws_queryModifyTargetGroupCommand, serializeAws_queryModifyTargetGroupCommand, } from "../protocols/Aws_query";
var ModifyTargetGroupCommand = (function (_super) {
    __extends(ModifyTargetGroupCommand, _super);
    function ModifyTargetGroupCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ModifyTargetGroupCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ElasticLoadBalancingV2Client";
        var commandName = "ModifyTargetGroupCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ModifyTargetGroupInput.filterSensitiveLog,
            outputFilterSensitiveLog: ModifyTargetGroupOutput.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ModifyTargetGroupCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryModifyTargetGroupCommand(input, context);
    };
    ModifyTargetGroupCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryModifyTargetGroupCommand(output, context);
    };
    return ModifyTargetGroupCommand;
}($Command));
export { ModifyTargetGroupCommand };
