// @aws-sdk/client-elastic-load-balancing-v2/dist-es/commands/RegisterTargetsCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { RegisterTargetsInput, RegisterTargetsOutput } from "../models/models_0";
import { deserializeAws_queryRegisterTargetsCommand, serializeAws_queryRegisterTargetsCommand, } from "../protocols/Aws_query";
var RegisterTargetsCommand = (function (_super) {
    __extends(RegisterTargetsCommand, _super);
    function RegisterTargetsCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    RegisterTargetsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ElasticLoadBalancingV2Client";
        var commandName = "RegisterTargetsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: RegisterTargetsInput.filterSensitiveLog,
            outputFilterSensitiveLog: RegisterTargetsOutput.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    RegisterTargetsCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryRegisterTargetsCommand(input, context);
    };
    RegisterTargetsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryRegisterTargetsCommand(output, context);
    };
    return RegisterTargetsCommand;
}($Command));
export { RegisterTargetsCommand };
