// @aws-sdk/client-elastic-load-balancing-v2/dist-es/commands/SetSecurityGroupsCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { SetSecurityGroupsInput, SetSecurityGroupsOutput } from "../models/models_0";
import { deserializeAws_querySetSecurityGroupsCommand, serializeAws_querySetSecurityGroupsCommand, } from "../protocols/Aws_query";
var SetSecurityGroupsCommand = (function (_super) {
    __extends(SetSecurityGroupsCommand, _super);
    function SetSecurityGroupsCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    SetSecurityGroupsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ElasticLoadBalancingV2Client";
        var commandName = "SetSecurityGroupsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: SetSecurityGroupsInput.filterSensitiveLog,
            outputFilterSensitiveLog: SetSecurityGroupsOutput.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    SetSecurityGroupsCommand.prototype.serialize = function (input, context) {
        return serializeAws_querySetSecurityGroupsCommand(input, context);
    };
    SetSecurityGroupsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_querySetSecurityGroupsCommand(output, context);
    };
    return SetSecurityGroupsCommand;
}($Command));
export { SetSecurityGroupsCommand };
