// @aws-sdk/client-elastic-load-balancing/dist-es/models/models_0.js
import { __assign } from "tslib";
export var AccessLog;
(function (AccessLog) {
    AccessLog.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(AccessLog || (AccessLog = {}));
export var AccessPointNotFoundException;
(function (AccessPointNotFoundException) {
    AccessPointNotFoundException.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(AccessPointNotFoundException || (AccessPointNotFoundException = {}));
export var AddAvailabilityZonesInput;
(function (AddAvailabilityZonesInput) {
    AddAvailabilityZonesInput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(AddAvailabilityZonesInput || (AddAvailabilityZonesInput = {}));
export var AddAvailabilityZonesOutput;
(function (AddAvailabilityZonesOutput) {
    AddAvailabilityZonesOutput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(AddAvailabilityZonesOutput || (AddAvailabilityZonesOutput = {}));
export var AdditionalAttribute;
(function (AdditionalAttribute) {
    AdditionalAttribute.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(AdditionalAttribute || (AdditionalAttribute = {}));
export var Tag;
(function (Tag) {
    Tag.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(Tag || (Tag = {}));
export var AddTagsInput;
(function (AddTagsInput) {
    AddTagsInput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(AddTagsInput || (AddTagsInput = {}));
export var AddTagsOutput;
(function (AddTagsOutput) {
    AddTagsOutput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(AddTagsOutput || (AddTagsOutput = {}));
export var DuplicateTagKeysException;
(function (DuplicateTagKeysException) {
    DuplicateTagKeysException.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DuplicateTagKeysException || (DuplicateTagKeysException = {}));
export var TooManyTagsException;
(function (TooManyTagsException) {
    TooManyTagsException.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(TooManyTagsException || (TooManyTagsException = {}));
export var AppCookieStickinessPolicy;
(function (AppCookieStickinessPolicy) {
    AppCookieStickinessPolicy.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(AppCookieStickinessPolicy || (AppCookieStickinessPolicy = {}));
export var ApplySecurityGroupsToLoadBalancerInput;
(function (ApplySecurityGroupsToLoadBalancerInput) {
    ApplySecurityGroupsToLoadBalancerInput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ApplySecurityGroupsToLoadBalancerInput || (ApplySecurityGroupsToLoadBalancerInput = {}));
export var ApplySecurityGroupsToLoadBalancerOutput;
(function (ApplySecurityGroupsToLoadBalancerOutput) {
    ApplySecurityGroupsToLoadBalancerOutput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ApplySecurityGroupsToLoadBalancerOutput || (ApplySecurityGroupsToLoadBalancerOutput = {}));
export var InvalidConfigurationRequestException;
(function (InvalidConfigurationRequestException) {
    InvalidConfigurationRequestException.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(InvalidConfigurationRequestException || (InvalidConfigurationRequestException = {}));
export var InvalidSecurityGroupException;
(function (InvalidSecurityGroupException) {
    InvalidSecurityGroupException.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(InvalidSecurityGroupException || (InvalidSecurityGroupException = {}));
export var AttachLoadBalancerToSubnetsInput;
(function (AttachLoadBalancerToSubnetsInput) {
    AttachLoadBalancerToSubnetsInput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(AttachLoadBalancerToSubnetsInput || (AttachLoadBalancerToSubnetsInput = {}));
export var AttachLoadBalancerToSubnetsOutput;
(function (AttachLoadBalancerToSubnetsOutput) {
    AttachLoadBalancerToSubnetsOutput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(AttachLoadBalancerToSubnetsOutput || (AttachLoadBalancerToSubnetsOutput = {}));
export var InvalidSubnetException;
(function (InvalidSubnetException) {
    InvalidSubnetException.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(InvalidSubnetException || (InvalidSubnetException = {}));
export var SubnetNotFoundException;
(function (SubnetNotFoundException) {
    SubnetNotFoundException.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(SubnetNotFoundException || (SubnetNotFoundException = {}));
export var BackendServerDescription;
(function (BackendServerDescription) {
    BackendServerDescription.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(BackendServerDescription || (BackendServerDescription = {}));
export var CertificateNotFoundException;
(function (CertificateNotFoundException) {
    CertificateNotFoundException.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CertificateNotFoundException || (CertificateNotFoundException = {}));
export var HealthCheck;
(function (HealthCheck) {
    HealthCheck.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(HealthCheck || (HealthCheck = {}));
export var ConfigureHealthCheckInput;
(function (ConfigureHealthCheckInput) {
    ConfigureHealthCheckInput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ConfigureHealthCheckInput || (ConfigureHealthCheckInput = {}));
export var ConfigureHealthCheckOutput;
(function (ConfigureHealthCheckOutput) {
    ConfigureHealthCheckOutput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ConfigureHealthCheckOutput || (ConfigureHealthCheckOutput = {}));
export var ConnectionDraining;
(function (ConnectionDraining) {
    ConnectionDraining.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ConnectionDraining || (ConnectionDraining = {}));
export var ConnectionSettings;
(function (ConnectionSettings) {
    ConnectionSettings.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ConnectionSettings || (ConnectionSettings = {}));
export var Listener;
(function (Listener) {
    Listener.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(Listener || (Listener = {}));
export var CreateAccessPointInput;
(function (CreateAccessPointInput) {
    CreateAccessPointInput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateAccessPointInput || (CreateAccessPointInput = {}));
export var CreateAccessPointOutput;
(function (CreateAccessPointOutput) {
    CreateAccessPointOutput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateAccessPointOutput || (CreateAccessPointOutput = {}));
export var CreateAppCookieStickinessPolicyInput;
(function (CreateAppCookieStickinessPolicyInput) {
    CreateAppCookieStickinessPolicyInput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateAppCookieStickinessPolicyInput || (CreateAppCookieStickinessPolicyInput = {}));
export var CreateAppCookieStickinessPolicyOutput;
(function (CreateAppCookieStickinessPolicyOutput) {
    CreateAppCookieStickinessPolicyOutput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateAppCookieStickinessPolicyOutput || (CreateAppCookieStickinessPolicyOutput = {}));
export var DuplicatePolicyNameException;
(function (DuplicatePolicyNameException) {
    DuplicatePolicyNameException.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DuplicatePolicyNameException || (DuplicatePolicyNameException = {}));
export var TooManyPoliciesException;
(function (TooManyPoliciesException) {
    TooManyPoliciesException.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(TooManyPoliciesException || (TooManyPoliciesException = {}));
export var CreateLBCookieStickinessPolicyInput;
(function (CreateLBCookieStickinessPolicyInput) {
    CreateLBCookieStickinessPolicyInput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateLBCookieStickinessPolicyInput || (CreateLBCookieStickinessPolicyInput = {}));
export var CreateLBCookieStickinessPolicyOutput;
(function (CreateLBCookieStickinessPolicyOutput) {
    CreateLBCookieStickinessPolicyOutput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateLBCookieStickinessPolicyOutput || (CreateLBCookieStickinessPolicyOutput = {}));
export var DuplicateAccessPointNameException;
(function (DuplicateAccessPointNameException) {
    DuplicateAccessPointNameException.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DuplicateAccessPointNameException || (DuplicateAccessPointNameException = {}));
export var InvalidSchemeException;
(function (InvalidSchemeException) {
    InvalidSchemeException.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(InvalidSchemeException || (InvalidSchemeException = {}));
export var OperationNotPermittedException;
(function (OperationNotPermittedException) {
    OperationNotPermittedException.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(OperationNotPermittedException || (OperationNotPermittedException = {}));
export var TooManyAccessPointsException;
(function (TooManyAccessPointsException) {
    TooManyAccessPointsException.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(TooManyAccessPointsException || (TooManyAccessPointsException = {}));
export var UnsupportedProtocolException;
(function (UnsupportedProtocolException) {
    UnsupportedProtocolException.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UnsupportedProtocolException || (UnsupportedProtocolException = {}));
export var CreateLoadBalancerListenerInput;
(function (CreateLoadBalancerListenerInput) {
    CreateLoadBalancerListenerInput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateLoadBalancerListenerInput || (CreateLoadBalancerListenerInput = {}));
export var CreateLoadBalancerListenerOutput;
(function (CreateLoadBalancerListenerOutput) {
    CreateLoadBalancerListenerOutput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateLoadBalancerListenerOutput || (CreateLoadBalancerListenerOutput = {}));
export var DuplicateListenerException;
(function (DuplicateListenerException) {
    DuplicateListenerException.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DuplicateListenerException || (DuplicateListenerException = {}));
export var PolicyAttribute;
(function (PolicyAttribute) {
    PolicyAttribute.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(PolicyAttribute || (PolicyAttribute = {}));
export var CreateLoadBalancerPolicyInput;
(function (CreateLoadBalancerPolicyInput) {
    CreateLoadBalancerPolicyInput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateLoadBalancerPolicyInput || (CreateLoadBalancerPolicyInput = {}));
export var CreateLoadBalancerPolicyOutput;
(function (CreateLoadBalancerPolicyOutput) {
    CreateLoadBalancerPolicyOutput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateLoadBalancerPolicyOutput || (CreateLoadBalancerPolicyOutput = {}));
export var PolicyTypeNotFoundException;
(function (PolicyTypeNotFoundException) {
    PolicyTypeNotFoundException.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(PolicyTypeNotFoundException || (PolicyTypeNotFoundException = {}));
export var CrossZoneLoadBalancing;
(function (CrossZoneLoadBalancing) {
    CrossZoneLoadBalancing.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CrossZoneLoadBalancing || (CrossZoneLoadBalancing = {}));
export var DeleteAccessPointInput;
(function (DeleteAccessPointInput) {
    DeleteAccessPointInput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteAccessPointInput || (DeleteAccessPointInput = {}));
export var DeleteAccessPointOutput;
(function (DeleteAccessPointOutput) {
    DeleteAccessPointOutput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteAccessPointOutput || (DeleteAccessPointOutput = {}));
export var DeleteLoadBalancerListenerInput;
(function (DeleteLoadBalancerListenerInput) {
    DeleteLoadBalancerListenerInput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteLoadBalancerListenerInput || (DeleteLoadBalancerListenerInput = {}));
export var DeleteLoadBalancerListenerOutput;
(function (DeleteLoadBalancerListenerOutput) {
    DeleteLoadBalancerListenerOutput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteLoadBalancerListenerOutput || (DeleteLoadBalancerListenerOutput = {}));
export var DeleteLoadBalancerPolicyInput;
(function (DeleteLoadBalancerPolicyInput) {
    DeleteLoadBalancerPolicyInput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteLoadBalancerPolicyInput || (DeleteLoadBalancerPolicyInput = {}));
export var DeleteLoadBalancerPolicyOutput;
(function (DeleteLoadBalancerPolicyOutput) {
    DeleteLoadBalancerPolicyOutput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteLoadBalancerPolicyOutput || (DeleteLoadBalancerPolicyOutput = {}));
export var DependencyThrottleException;
(function (DependencyThrottleException) {
    DependencyThrottleException.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DependencyThrottleException || (DependencyThrottleException = {}));
export var Instance;
(function (Instance) {
    Instance.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(Instance || (Instance = {}));
export var DeregisterEndPointsInput;
(function (DeregisterEndPointsInput) {
    DeregisterEndPointsInput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeregisterEndPointsInput || (DeregisterEndPointsInput = {}));
export var DeregisterEndPointsOutput;
(function (DeregisterEndPointsOutput) {
    DeregisterEndPointsOutput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeregisterEndPointsOutput || (DeregisterEndPointsOutput = {}));
export var InvalidEndPointException;
(function (InvalidEndPointException) {
    InvalidEndPointException.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(InvalidEndPointException || (InvalidEndPointException = {}));
export var DescribeAccessPointsInput;
(function (DescribeAccessPointsInput) {
    DescribeAccessPointsInput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeAccessPointsInput || (DescribeAccessPointsInput = {}));
export var ListenerDescription;
(function (ListenerDescription) {
    ListenerDescription.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListenerDescription || (ListenerDescription = {}));
export var LBCookieStickinessPolicy;
(function (LBCookieStickinessPolicy) {
    LBCookieStickinessPolicy.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(LBCookieStickinessPolicy || (LBCookieStickinessPolicy = {}));
export var Policies;
(function (Policies) {
    Policies.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(Policies || (Policies = {}));
export var SourceSecurityGroup;
(function (SourceSecurityGroup) {
    SourceSecurityGroup.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(SourceSecurityGroup || (SourceSecurityGroup = {}));
export var LoadBalancerDescription;
(function (LoadBalancerDescription) {
    LoadBalancerDescription.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(LoadBalancerDescription || (LoadBalancerDescription = {}));
export var DescribeAccessPointsOutput;
(function (DescribeAccessPointsOutput) {
    DescribeAccessPointsOutput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeAccessPointsOutput || (DescribeAccessPointsOutput = {}));
export var DescribeAccountLimitsInput;
(function (DescribeAccountLimitsInput) {
    DescribeAccountLimitsInput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeAccountLimitsInput || (DescribeAccountLimitsInput = {}));
export var Limit;
(function (Limit) {
    Limit.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(Limit || (Limit = {}));
export var DescribeAccountLimitsOutput;
(function (DescribeAccountLimitsOutput) {
    DescribeAccountLimitsOutput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeAccountLimitsOutput || (DescribeAccountLimitsOutput = {}));
export var DescribeEndPointStateInput;
(function (DescribeEndPointStateInput) {
    DescribeEndPointStateInput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeEndPointStateInput || (DescribeEndPointStateInput = {}));
export var InstanceState;
(function (InstanceState) {
    InstanceState.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(InstanceState || (InstanceState = {}));
export var DescribeEndPointStateOutput;
(function (DescribeEndPointStateOutput) {
    DescribeEndPointStateOutput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeEndPointStateOutput || (DescribeEndPointStateOutput = {}));
export var DescribeLoadBalancerAttributesInput;
(function (DescribeLoadBalancerAttributesInput) {
    DescribeLoadBalancerAttributesInput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeLoadBalancerAttributesInput || (DescribeLoadBalancerAttributesInput = {}));
export var LoadBalancerAttributes;
(function (LoadBalancerAttributes) {
    LoadBalancerAttributes.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(LoadBalancerAttributes || (LoadBalancerAttributes = {}));
export var DescribeLoadBalancerAttributesOutput;
(function (DescribeLoadBalancerAttributesOutput) {
    DescribeLoadBalancerAttributesOutput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeLoadBalancerAttributesOutput || (DescribeLoadBalancerAttributesOutput = {}));
export var LoadBalancerAttributeNotFoundException;
(function (LoadBalancerAttributeNotFoundException) {
    LoadBalancerAttributeNotFoundException.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(LoadBalancerAttributeNotFoundException || (LoadBalancerAttributeNotFoundException = {}));
export var DescribeLoadBalancerPoliciesInput;
(function (DescribeLoadBalancerPoliciesInput) {
    DescribeLoadBalancerPoliciesInput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeLoadBalancerPoliciesInput || (DescribeLoadBalancerPoliciesInput = {}));
export var PolicyAttributeDescription;
(function (PolicyAttributeDescription) {
    PolicyAttributeDescription.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(PolicyAttributeDescription || (PolicyAttributeDescription = {}));
export var PolicyDescription;
(function (PolicyDescription) {
    PolicyDescription.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(PolicyDescription || (PolicyDescription = {}));
export var DescribeLoadBalancerPoliciesOutput;
(function (DescribeLoadBalancerPoliciesOutput) {
    DescribeLoadBalancerPoliciesOutput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeLoadBalancerPoliciesOutput || (DescribeLoadBalancerPoliciesOutput = {}));
export var PolicyNotFoundException;
(function (PolicyNotFoundException) {
    PolicyNotFoundException.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(PolicyNotFoundException || (PolicyNotFoundException = {}));
export var DescribeLoadBalancerPolicyTypesInput;
(function (DescribeLoadBalancerPolicyTypesInput) {
    DescribeLoadBalancerPolicyTypesInput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeLoadBalancerPolicyTypesInput || (DescribeLoadBalancerPolicyTypesInput = {}));
export var PolicyAttributeTypeDescription;
(function (PolicyAttributeTypeDescription) {
    PolicyAttributeTypeDescription.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(PolicyAttributeTypeDescription || (PolicyAttributeTypeDescription = {}));
export var PolicyTypeDescription;
(function (PolicyTypeDescription) {
    PolicyTypeDescription.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(PolicyTypeDescription || (PolicyTypeDescription = {}));
export var DescribeLoadBalancerPolicyTypesOutput;
(function (DescribeLoadBalancerPolicyTypesOutput) {
    DescribeLoadBalancerPolicyTypesOutput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeLoadBalancerPolicyTypesOutput || (DescribeLoadBalancerPolicyTypesOutput = {}));
export var DescribeTagsInput;
(function (DescribeTagsInput) {
    DescribeTagsInput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeTagsInput || (DescribeTagsInput = {}));
export var TagDescription;
(function (TagDescription) {
    TagDescription.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(TagDescription || (TagDescription = {}));
export var DescribeTagsOutput;
(function (DescribeTagsOutput) {
    DescribeTagsOutput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeTagsOutput || (DescribeTagsOutput = {}));
export var DetachLoadBalancerFromSubnetsInput;
(function (DetachLoadBalancerFromSubnetsInput) {
    DetachLoadBalancerFromSubnetsInput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DetachLoadBalancerFromSubnetsInput || (DetachLoadBalancerFromSubnetsInput = {}));
export var DetachLoadBalancerFromSubnetsOutput;
(function (DetachLoadBalancerFromSubnetsOutput) {
    DetachLoadBalancerFromSubnetsOutput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DetachLoadBalancerFromSubnetsOutput || (DetachLoadBalancerFromSubnetsOutput = {}));
export var RemoveAvailabilityZonesInput;
(function (RemoveAvailabilityZonesInput) {
    RemoveAvailabilityZonesInput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(RemoveAvailabilityZonesInput || (RemoveAvailabilityZonesInput = {}));
export var RemoveAvailabilityZonesOutput;
(function (RemoveAvailabilityZonesOutput) {
    RemoveAvailabilityZonesOutput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(RemoveAvailabilityZonesOutput || (RemoveAvailabilityZonesOutput = {}));
export var ModifyLoadBalancerAttributesInput;
(function (ModifyLoadBalancerAttributesInput) {
    ModifyLoadBalancerAttributesInput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ModifyLoadBalancerAttributesInput || (ModifyLoadBalancerAttributesInput = {}));
export var ModifyLoadBalancerAttributesOutput;
(function (ModifyLoadBalancerAttributesOutput) {
    ModifyLoadBalancerAttributesOutput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ModifyLoadBalancerAttributesOutput || (ModifyLoadBalancerAttributesOutput = {}));
export var RegisterEndPointsInput;
(function (RegisterEndPointsInput) {
    RegisterEndPointsInput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(RegisterEndPointsInput || (RegisterEndPointsInput = {}));
export var RegisterEndPointsOutput;
(function (RegisterEndPointsOutput) {
    RegisterEndPointsOutput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(RegisterEndPointsOutput || (RegisterEndPointsOutput = {}));
export var TagKeyOnly;
(function (TagKeyOnly) {
    TagKeyOnly.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(TagKeyOnly || (TagKeyOnly = {}));
export var RemoveTagsInput;
(function (RemoveTagsInput) {
    RemoveTagsInput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(RemoveTagsInput || (RemoveTagsInput = {}));
export var RemoveTagsOutput;
(function (RemoveTagsOutput) {
    RemoveTagsOutput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(RemoveTagsOutput || (RemoveTagsOutput = {}));
export var ListenerNotFoundException;
(function (ListenerNotFoundException) {
    ListenerNotFoundException.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListenerNotFoundException || (ListenerNotFoundException = {}));
export var SetLoadBalancerListenerSSLCertificateInput;
(function (SetLoadBalancerListenerSSLCertificateInput) {
    SetLoadBalancerListenerSSLCertificateInput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(SetLoadBalancerListenerSSLCertificateInput || (SetLoadBalancerListenerSSLCertificateInput = {}));
export var SetLoadBalancerListenerSSLCertificateOutput;
(function (SetLoadBalancerListenerSSLCertificateOutput) {
    SetLoadBalancerListenerSSLCertificateOutput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(SetLoadBalancerListenerSSLCertificateOutput || (SetLoadBalancerListenerSSLCertificateOutput = {}));
export var SetLoadBalancerPoliciesForBackendServerInput;
(function (SetLoadBalancerPoliciesForBackendServerInput) {
    SetLoadBalancerPoliciesForBackendServerInput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(SetLoadBalancerPoliciesForBackendServerInput || (SetLoadBalancerPoliciesForBackendServerInput = {}));
export var SetLoadBalancerPoliciesForBackendServerOutput;
(function (SetLoadBalancerPoliciesForBackendServerOutput) {
    SetLoadBalancerPoliciesForBackendServerOutput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(SetLoadBalancerPoliciesForBackendServerOutput || (SetLoadBalancerPoliciesForBackendServerOutput = {}));
export var SetLoadBalancerPoliciesOfListenerInput;
(function (SetLoadBalancerPoliciesOfListenerInput) {
    SetLoadBalancerPoliciesOfListenerInput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(SetLoadBalancerPoliciesOfListenerInput || (SetLoadBalancerPoliciesOfListenerInput = {}));
export var SetLoadBalancerPoliciesOfListenerOutput;
(function (SetLoadBalancerPoliciesOfListenerOutput) {
    SetLoadBalancerPoliciesOfListenerOutput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(SetLoadBalancerPoliciesOfListenerOutput || (SetLoadBalancerPoliciesOfListenerOutput = {}));
