// @aws-sdk/client-acm/dist-es/commands/AddTagsToCertificateCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { AddTagsToCertificateRequest } from "../models/models_0";
import { deserializeAws_json1_1AddTagsToCertificateCommand, serializeAws_json1_1AddTagsToCertificateCommand, } from "../protocols/Aws_json1_1";
var AddTagsToCertificateCommand = (function (_super) {
    __extends(AddTagsToCertificateCommand, _super);
    function AddTagsToCertificateCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    AddTagsToCertificateCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ACMClient";
        var commandName = "AddTagsToCertificateCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: AddTagsToCertificateRequest.filterSensitiveLog,
            outputFilterSensitiveLog: function (output) { return output; },
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    AddTagsToCertificateCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1AddTagsToCertificateCommand(input, context);
    };
    AddTagsToCertificateCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1AddTagsToCertificateCommand(output, context);
    };
    return AddTagsToCertificateCommand;
}($Command));
export { AddTagsToCertificateCommand };
