// @aws-sdk/client-acm/dist-es/commands/ExportCertificateCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ExportCertificateRequest, ExportCertificateResponse } from "../models/models_0";
import { deserializeAws_json1_1ExportCertificateCommand, serializeAws_json1_1ExportCertificateCommand, } from "../protocols/Aws_json1_1";
var ExportCertificateCommand = (function (_super) {
    __extends(ExportCertificateCommand, _super);
    function ExportCertificateCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ExportCertificateCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ACMClient";
        var commandName = "ExportCertificateCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ExportCertificateRequest.filterSensitiveLog,
            outputFilterSensitiveLog: ExportCertificateResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ExportCertificateCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1ExportCertificateCommand(input, context);
    };
    ExportCertificateCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1ExportCertificateCommand(output, context);
    };
    return ExportCertificateCommand;
}($Command));
export { ExportCertificateCommand };
