// @aws-sdk/client-acm/dist-es/commands/GetAccountConfigurationCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { GetAccountConfigurationResponse } from "../models/models_0";
import { deserializeAws_json1_1GetAccountConfigurationCommand, serializeAws_json1_1GetAccountConfigurationCommand, } from "../protocols/Aws_json1_1";
var GetAccountConfigurationCommand = (function (_super) {
    __extends(GetAccountConfigurationCommand, _super);
    function GetAccountConfigurationCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    GetAccountConfigurationCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ACMClient";
        var commandName = "GetAccountConfigurationCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: function (input) { return input; },
            outputFilterSensitiveLog: GetAccountConfigurationResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    GetAccountConfigurationCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1GetAccountConfigurationCommand(input, context);
    };
    GetAccountConfigurationCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1GetAccountConfigurationCommand(output, context);
    };
    return GetAccountConfigurationCommand;
}($Command));
export { GetAccountConfigurationCommand };
