// @aws-sdk/client-acm/dist-es/commands/ListTagsForCertificateCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ListTagsForCertificateRequest, ListTagsForCertificateResponse } from "../models/models_0";
import { deserializeAws_json1_1ListTagsForCertificateCommand, serializeAws_json1_1ListTagsForCertificateCommand, } from "../protocols/Aws_json1_1";
var ListTagsForCertificateCommand = (function (_super) {
    __extends(ListTagsForCertificateCommand, _super);
    function ListTagsForCertificateCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ListTagsForCertificateCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ACMClient";
        var commandName = "ListTagsForCertificateCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ListTagsForCertificateRequest.filterSensitiveLog,
            outputFilterSensitiveLog: ListTagsForCertificateResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ListTagsForCertificateCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1ListTagsForCertificateCommand(input, context);
    };
    ListTagsForCertificateCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1ListTagsForCertificateCommand(output, context);
    };
    return ListTagsForCertificateCommand;
}($Command));
export { ListTagsForCertificateCommand };
