// @aws-sdk/client-elastic-load-balancing/dist-es/commands/CreateLoadBalancerListenersCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { CreateLoadBalancerListenerInput, CreateLoadBalancerListenerOutput } from "../models/models_0";
import { deserializeAws_queryCreateLoadBalancerListenersCommand, serializeAws_queryCreateLoadBalancerListenersCommand, } from "../protocols/Aws_query";
var CreateLoadBalancerListenersCommand = (function (_super) {
    __extends(CreateLoadBalancerListenersCommand, _super);
    function CreateLoadBalancerListenersCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    CreateLoadBalancerListenersCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ElasticLoadBalancingClient";
        var commandName = "CreateLoadBalancerListenersCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: CreateLoadBalancerListenerInput.filterSensitiveLog,
            outputFilterSensitiveLog: CreateLoadBalancerListenerOutput.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    CreateLoadBalancerListenersCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryCreateLoadBalancerListenersCommand(input, context);
    };
    CreateLoadBalancerListenersCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryCreateLoadBalancerListenersCommand(output, context);
    };
    return CreateLoadBalancerListenersCommand;
}($Command));
export { CreateLoadBalancerListenersCommand };
