// @aws-sdk/client-elastic-load-balancing/dist-es/commands/CreateLBCookieStickinessPolicyCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { CreateLBCookieStickinessPolicyInput, CreateLBCookieStickinessPolicyOutput } from "../models/models_0";
import { deserializeAws_queryCreateLBCookieStickinessPolicyCommand, serializeAws_queryCreateLBCookieStickinessPolicyCommand, } from "../protocols/Aws_query";
var CreateLBCookieStickinessPolicyCommand = (function (_super) {
    __extends(CreateLBCookieStickinessPolicyCommand, _super);
    function CreateLBCookieStickinessPolicyCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    CreateLBCookieStickinessPolicyCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ElasticLoadBalancingClient";
        var commandName = "CreateLBCookieStickinessPolicyCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: CreateLBCookieStickinessPolicyInput.filterSensitiveLog,
            outputFilterSensitiveLog: CreateLBCookieStickinessPolicyOutput.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    CreateLBCookieStickinessPolicyCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryCreateLBCookieStickinessPolicyCommand(input, context);
    };
    CreateLBCookieStickinessPolicyCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryCreateLBCookieStickinessPolicyCommand(output, context);
    };
    return CreateLBCookieStickinessPolicyCommand;
}($Command));
export { CreateLBCookieStickinessPolicyCommand };
