// @aws-sdk/client-elastic-load-balancing/dist-es/commands/DescribeLoadBalancerPoliciesCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DescribeLoadBalancerPoliciesInput, DescribeLoadBalancerPoliciesOutput } from "../models/models_0";
import { deserializeAws_queryDescribeLoadBalancerPoliciesCommand, serializeAws_queryDescribeLoadBalancerPoliciesCommand, } from "../protocols/Aws_query";
var DescribeLoadBalancerPoliciesCommand = (function (_super) {
    __extends(DescribeLoadBalancerPoliciesCommand, _super);
    function DescribeLoadBalancerPoliciesCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DescribeLoadBalancerPoliciesCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ElasticLoadBalancingClient";
        var commandName = "DescribeLoadBalancerPoliciesCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DescribeLoadBalancerPoliciesInput.filterSensitiveLog,
            outputFilterSensitiveLog: DescribeLoadBalancerPoliciesOutput.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DescribeLoadBalancerPoliciesCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryDescribeLoadBalancerPoliciesCommand(input, context);
    };
    DescribeLoadBalancerPoliciesCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryDescribeLoadBalancerPoliciesCommand(output, context);
    };
    return DescribeLoadBalancerPoliciesCommand;
}($Command));
export { DescribeLoadBalancerPoliciesCommand };
