// @aws-sdk/client-elastic-load-balancing/dist-es/commands/DescribeInstanceHealthCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DescribeEndPointStateInput, DescribeEndPointStateOutput } from "../models/models_0";
import { deserializeAws_queryDescribeInstanceHealthCommand, serializeAws_queryDescribeInstanceHealthCommand, } from "../protocols/Aws_query";
var DescribeInstanceHealthCommand = (function (_super) {
    __extends(DescribeInstanceHealthCommand, _super);
    function DescribeInstanceHealthCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DescribeInstanceHealthCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ElasticLoadBalancingClient";
        var commandName = "DescribeInstanceHealthCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DescribeEndPointStateInput.filterSensitiveLog,
            outputFilterSensitiveLog: DescribeEndPointStateOutput.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DescribeInstanceHealthCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryDescribeInstanceHealthCommand(input, context);
    };
    DescribeInstanceHealthCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryDescribeInstanceHealthCommand(output, context);
    };
    return DescribeInstanceHealthCommand;
}($Command));
export { DescribeInstanceHealthCommand };
