// @aws-sdk/client-elastic-load-balancing/dist-es/commands/DescribeLoadBalancerPolicyTypesCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DescribeLoadBalancerPolicyTypesInput, DescribeLoadBalancerPolicyTypesOutput } from "../models/models_0";
import { deserializeAws_queryDescribeLoadBalancerPolicyTypesCommand, serializeAws_queryDescribeLoadBalancerPolicyTypesCommand, } from "../protocols/Aws_query";
var DescribeLoadBalancerPolicyTypesCommand = (function (_super) {
    __extends(DescribeLoadBalancerPolicyTypesCommand, _super);
    function DescribeLoadBalancerPolicyTypesCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DescribeLoadBalancerPolicyTypesCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ElasticLoadBalancingClient";
        var commandName = "DescribeLoadBalancerPolicyTypesCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DescribeLoadBalancerPolicyTypesInput.filterSensitiveLog,
            outputFilterSensitiveLog: DescribeLoadBalancerPolicyTypesOutput.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DescribeLoadBalancerPolicyTypesCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryDescribeLoadBalancerPolicyTypesCommand(input, context);
    };
    DescribeLoadBalancerPolicyTypesCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryDescribeLoadBalancerPolicyTypesCommand(output, context);
    };
    return DescribeLoadBalancerPolicyTypesCommand;
}($Command));
export { DescribeLoadBalancerPolicyTypesCommand };
