// @aws-sdk/client-elastic-load-balancing/dist-es/commands/DisableAvailabilityZonesForLoadBalancerCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { RemoveAvailabilityZonesInput, RemoveAvailabilityZonesOutput } from "../models/models_0";
import { deserializeAws_queryDisableAvailabilityZonesForLoadBalancerCommand, serializeAws_queryDisableAvailabilityZonesForLoadBalancerCommand, } from "../protocols/Aws_query";
var DisableAvailabilityZonesForLoadBalancerCommand = (function (_super) {
    __extends(DisableAvailabilityZonesForLoadBalancerCommand, _super);
    function DisableAvailabilityZonesForLoadBalancerCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DisableAvailabilityZonesForLoadBalancerCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ElasticLoadBalancingClient";
        var commandName = "DisableAvailabilityZonesForLoadBalancerCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: RemoveAvailabilityZonesInput.filterSensitiveLog,
            outputFilterSensitiveLog: RemoveAvailabilityZonesOutput.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DisableAvailabilityZonesForLoadBalancerCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryDisableAvailabilityZonesForLoadBalancerCommand(input, context);
    };
    DisableAvailabilityZonesForLoadBalancerCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryDisableAvailabilityZonesForLoadBalancerCommand(output, context);
    };
    return DisableAvailabilityZonesForLoadBalancerCommand;
}($Command));
export { DisableAvailabilityZonesForLoadBalancerCommand };
