// @aws-sdk/client-elastic-load-balancing/dist-es/commands/SetLoadBalancerPoliciesForBackendServerCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { SetLoadBalancerPoliciesForBackendServerInput, SetLoadBalancerPoliciesForBackendServerOutput, } from "../models/models_0";
import { deserializeAws_querySetLoadBalancerPoliciesForBackendServerCommand, serializeAws_querySetLoadBalancerPoliciesForBackendServerCommand, } from "../protocols/Aws_query";
var SetLoadBalancerPoliciesForBackendServerCommand = (function (_super) {
    __extends(SetLoadBalancerPoliciesForBackendServerCommand, _super);
    function SetLoadBalancerPoliciesForBackendServerCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    SetLoadBalancerPoliciesForBackendServerCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ElasticLoadBalancingClient";
        var commandName = "SetLoadBalancerPoliciesForBackendServerCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: SetLoadBalancerPoliciesForBackendServerInput.filterSensitiveLog,
            outputFilterSensitiveLog: SetLoadBalancerPoliciesForBackendServerOutput.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    SetLoadBalancerPoliciesForBackendServerCommand.prototype.serialize = function (input, context) {
        return serializeAws_querySetLoadBalancerPoliciesForBackendServerCommand(input, context);
    };
    SetLoadBalancerPoliciesForBackendServerCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_querySetLoadBalancerPoliciesForBackendServerCommand(output, context);
    };
    return SetLoadBalancerPoliciesForBackendServerCommand;
}($Command));
export { SetLoadBalancerPoliciesForBackendServerCommand };
