// @aws-sdk/client-elastic-load-balancing/dist-es/commands/SetLoadBalancerPoliciesOfListenerCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { SetLoadBalancerPoliciesOfListenerInput, SetLoadBalancerPoliciesOfListenerOutput } from "../models/models_0";
import { deserializeAws_querySetLoadBalancerPoliciesOfListenerCommand, serializeAws_querySetLoadBalancerPoliciesOfListenerCommand, } from "../protocols/Aws_query";
var SetLoadBalancerPoliciesOfListenerCommand = (function (_super) {
    __extends(SetLoadBalancerPoliciesOfListenerCommand, _super);
    function SetLoadBalancerPoliciesOfListenerCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    SetLoadBalancerPoliciesOfListenerCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ElasticLoadBalancingClient";
        var commandName = "SetLoadBalancerPoliciesOfListenerCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: SetLoadBalancerPoliciesOfListenerInput.filterSensitiveLog,
            outputFilterSensitiveLog: SetLoadBalancerPoliciesOfListenerOutput.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    SetLoadBalancerPoliciesOfListenerCommand.prototype.serialize = function (input, context) {
        return serializeAws_querySetLoadBalancerPoliciesOfListenerCommand(input, context);
    };
    SetLoadBalancerPoliciesOfListenerCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_querySetLoadBalancerPoliciesOfListenerCommand(output, context);
    };
    return SetLoadBalancerPoliciesOfListenerCommand;
}($Command));
export { SetLoadBalancerPoliciesOfListenerCommand };
