// @aws-sdk/client-elastic-load-balancing/dist-cjs/commands/AttachLoadBalancerToSubnetsCommand.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AttachLoadBalancerToSubnetsCommand = void 0;
const middleware_serde_1 = require("@aws-sdk/middleware-serde");
const smithy_client_1 = require("@aws-sdk/smithy-client");
const models_0_1 = require("../models/models_0");
const Aws_query_1 = require("../protocols/Aws_query");
class AttachLoadBalancerToSubnetsCommand extends smithy_client_1.Command {
    constructor(input) {
        super();
        this.input = input;
    }
    resolveMiddleware(clientStack, configuration, options) {
        this.middlewareStack.use(middleware_serde_1.getSerdePlugin(configuration, this.serialize, this.deserialize));
        const stack = clientStack.concat(this.middlewareStack);
        const { logger } = configuration;
        const clientName = "ElasticLoadBalancingClient";
        const commandName = "AttachLoadBalancerToSubnetsCommand";
        const handlerExecutionContext = {
            logger,
            clientName,
            commandName,
            inputFilterSensitiveLog: models_0_1.AttachLoadBalancerToSubnetsInput.filterSensitiveLog,
            outputFilterSensitiveLog: models_0_1.AttachLoadBalancerToSubnetsOutput.filterSensitiveLog,
        };
        const { requestHandler } = configuration;
        return stack.resolve((request) => requestHandler.handle(request.request, options || {}), handlerExecutionContext);
    }
    serialize(input, context) {
        return Aws_query_1.serializeAws_queryAttachLoadBalancerToSubnetsCommand(input, context);
    }
    deserialize(output, context) {
        return Aws_query_1.deserializeAws_queryAttachLoadBalancerToSubnetsCommand(output, context);
    }
}
exports.AttachLoadBalancerToSubnetsCommand = AttachLoadBalancerToSubnetsCommand;
